### PR TITLE
add function to split patches at knots based on multiplicity

### PR DESCRIPTION
**Pull request by filiatra**

### CODE REVIEW

File: src/gsCore/gsGeometry.h [hunk 1]
```diff
@@ -506,22 +506,30 @@ class gsGeometry : public gsFunction<T>
     }
 
     /// Embeds coefficients in \a N dimension
-    void embed(index_t N)
-    { 
+    void embed(index_t N, bool lr = true)
```
Comment (stefantakacs) [inline src/gsCore/gsGeometry.h]: What does lr mean? Maybe add to doxygen comment.

File: src/gsUtils/gsMesh/gsVertex.h [hunk 2]
```diff
@@ -14,7 +14,7 @@
 #pragma once
 
 #include <gsUtils/gsMesh/gsMeshElement.h>
-#include <gsCore/gsLinearAlgebra.h>
+//#include <gsCore/gsLinearAlgebra.h>
```
Comment (stefantakacs) [inline src/gsUtils/gsMesh/gsVertex.h]: I think we should not comment such things out but just remove them. The same holds for the things below.

File: src/gsIO/gsWriteParaview.hpp [hunk 3]
```diff
@@ -489,6 +489,12 @@ void writeSingleGeometry(gsFunction<T> const& func,
 
         if ( n == 1 )
         {
+            if (d==3)
```
Comment (stefantakacs) [inline src/gsIO/gsWriteParaview.hpp:492]: This one I do not understand.

Comment (filiatra) [reply src/gsIO/gsWriteParaview.hpp:492]: In here we have n==1, so we plot a scalar function on its domain.
When the domain is less than 3, the graph is shown in 3D.
The new if is for a scalar function on a 3D domain: the scalar field will be shown as color over the 3D domain.

Comment (stefantakacs) [reply src/gsIO/gsWriteParaview.hpp:492]: Thanks

**Review by stefantakacs — APPROVED** [review]

**Pull request merged**